packages/execution-environments/src/common/endowments/timeout.ts

```diff
@@ -18,25 +18,27 @@ const createTimeout = () => {
       );
     }
 
-    const handle = setTimeout(() => {
-      registeredTimeouts.delete(handle);
+    const handle = Object.freeze({});
```

Comment:
Just curious, how is a frozen empty object enough to use as a key 🤔 Does it use the reference as the key?

Reply:
Yup, the reference. And empty object to keep it opaque.